File: 257-binary-tree-paths/binary-tree-paths.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void helper(vector<string>& result, TreeNode* root, string path = ""){
        if(root == nullptr){
            return;
        }

        if(root->left == nullptr && root->right == nullptr){
            path += to_string(root->val);
            result.push_back(path);
            path = "";
        }

        path += to_string(root->val) + "->";
        helper(result, root->left, path);
        helper(result, root->right, path);
    }

    vector<string> binaryTreePaths(TreeNode* root) {
        vector<string> result;
        helper(result, root);
        return result;
    }
};

File: 257-binary-tree-paths/binary-tree-paths.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private void helper(TreeNode root, List<String> result, String path){
        if(root == null){
            return;
        }

        if(root.left == null && root.right == null){
            path+= Integer.toString(root.val);
            result.add(path);
            path = "";
        }

        path+= Integer.toString(root.val) + "->";
        helper(root.left, result, path);
        helper(root.right,result,path);
    }

    public List<String> binaryTreePaths(TreeNode root) {
        List<String> result = new ArrayList<>();
        String path = "";
        helper(root,result, path);
        return result;
    }
}